salt/modules/highstate_doc.py

```diff
@@ -652,25 +652,26 @@ def proccesser_markdown(lowstate_item, config, **kwargs):
 
     # TODO: use salt defined STATE_REQUISITE_IN_KEYWORDS
     requisites = ""
+    next_w_item = next(iter(w.items()))
     if s.get("watch"):
         requisites += "run or update after changes in:\n"
         for w in s.get("watch", []):
-            requisites += _format_markdown_requisite(w.items()[0][0], w.items()[0][1])
+            requisites += _format_markdown_requisite(next_w_item[0], next_w_item[1],)
```

Comment:
`_format_markdown_requisite(next_w_item[0], next_w_item[1],)` is actually called itself up to 4 times, should that instead be the thing moved up?

Reply:
I see there are 4 almost equal cases that looks like could be shrunken to one case.
Everything We could do something like this here:
```python
for comment, key in (("run or update after changes in:\n", "watch"), ("after changes, run or update:\n", "watch_in"), ("require:\n", "require"), ("required in:\n", "require_in")):
    reqs = s.get(key, [])
    if reqs:
        requisites += comment
        for w in reqs:
            requisites += _format_markdown_requisite(*next(iter(w.items())))
```

Reply:
Ugh how did I not see that. Thanks :) done